### PR TITLE
Optional "block" given to RPC "work_generate" to infer difficulty

### PR DESCRIPTION
**Pull request by SergiySW**
https://github.com/nanocurrency/nano-node/pull/2754

### CODE REVIEW

File: docs/commands/rpc-protocol.md [hunk 1]
```diff
@@ -2787,9 +2787,14 @@ A valid Nano account. If provided and `use_peers` is set to `true`, this informa
...
 **Optional "version"**
 
-_version 21.0+_
+_version 21.0+_  
 Work version string. Currently "work_1" is the default and only valid option.
 
+**Optional "block"**
+
+_version 21.0+_  
+A valid Nano block (string or JSON). Using the optional `json_block` is recommended. If provided and no `difficulty` or `multiplier` are set, is used to calculate work generation difficulty for given block from node ledger.
```
Comment (guilhermelawless) [inline docs/commands/rpc-protocol.md:2796]: Should we use a similar description to RPC block_create optional "difficulty"?

Comment (SergiySW) [reply docs/commands/rpc-protocol.md:2796]: We have "If difficulty and work values are both not given, RPC processor tries to calculate difficulty for work generation based on ledger data: epoch from previous block or from link for receive subtype; block subtype from previous block balance."

Comment (guilhermelawless) [reply docs/commands/rpc-protocol.md:2796]:
```suggestion
A valid Nano block (string or JSON). Using the optional `json_block` is recommended. If provided and `difficulty` or `multiplier` are both not given, RPC processor tries to calculate the approprriate difficulty threshold based on ledger data.
```

Like this maybe?

Could you also fix the lack of _newline_ in `block_create` ? See image
image

**Comment by guilhermelawless** [comment]
Seems to be missing `json_block` option.

**Pull request merged**